hubblestack/extmods/fdg/stat.py

```diff
@@ -0,0 +1,250 @@
+                if not subcheck_passed:
+                    passed = False
+                    reason = {'expected': str(expected[attribute]),
+                              'allow_more_strict': str(allow_more_strict),
+                              'current': str(file_attribute_value)}
+                    reason_dict[attribute] = reason
+        else:
+            subcheck_passed = (str(expected[attribute]) == str(file_attribute_value))
+            if not subcheck_passed:
+                passed = False
+                reason = {'expected': str(expected[attribute]),
+                          'current': str(file_attribute_value)}
+                reason_dict[attribute] = reason
+
+    if reason_dict:
+        ret = {'Failure': "For file '{0}': {1}".format(filepath, reason_dict), "expected": expected}
+
+    if passed:
+        ret = {'Success': 'all stats matching', "expected": expected}
+        log.info("FDG stat is returning status : True, value : {0}".format(ret))
+        return True, ret
+    else:
+        log.info("FDG stat is returning status : False, value : {0}".format(ret))
+        return False, ret
+
+
+def _check_corner_cases(filepath, expected):
+    if not expected:
+        ret = {"Success" : "nothing is expected, therefore passing the test", "expected": expected}
+        log.info("FDG stat is returning status : True, value : {0}".format(ret))
+        return True, ret
+    elif 'match_on_file_missing' in expected.keys() and expected['match_on_file_missing']:
+        ret = {"Success": "unable to find file, passing check because 'match_on_file_missing' is 'True'", "expected": expected}
+        log.info("FDG stat is returning status : True, value : {0}".format(ret))
+        return True, ret
+    else:
+        reason = "Could not get access any file at '{0}'. " \
+                 "File might not exist, or hubble might not" \
+                 " have enough permissions".format(filepath)
+        ret = {'Failure': reason, "expected": expected}
+        log.info("FDG stat is returning status : False, value : {0}".format(ret))
+        return False, ret
+
+
+def _validate_inputs(filepath, expected):
+    ret = ''
+    if filepath is None or filepath == '':
+        log.error("filepath not specified")
+        ret = {'Failure': "no filepath provided", "expected": expected}
+        log.info("FDG stat is returning status : True, value : {0}".format(ret))
+        return False, ret
+
+    if 'allow_more_strict' in expected.keys() and 'mode' not in expected.keys():
+        reason = "'allow_more_strict' tag can't be specified without 'mode' tag." \
+                 " Seems like a bug in hubble profile."
+        ret = {'Failure': reason, "expected": expected}
+        log.info("FDG stat is returning status : True, value : {0}".format(ret))
+        return False, ret
+
+    return True, ret
+
+def _check_mode(max_permission, given_permission, allow_more_strict):
```

Comment:
This is exactly duplicate of stat module code.

We should put this code at a single place. In some common file. (utils/fs_commons.py)
And, should use that at both places.

Reply:
@goravsingal agree.

Reply:
Moved to a new util. Changed the code at both the places. Also moved the corresponding test cases to new file.